projects/storefrontlib/src/cms-components/organization/units/unit-tree-navigation-ui/unit-tree-navigation-ui.component.spec.ts

```diff
@@ -0,0 +1,212 @@
+    element = fixture.debugElement;
+    component.node = mockNode;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('executing constructor and calling breakpoint service', async () => {
+    beforeEach(() => {
+      spyOnProperty(breakpointService, 'isDown').and.returnValue(
+        of(BREAKPOINT.md)
+      );
+    });
+  });
+
+  it('should trigger refreshUIMappedElements & mapUlElementToExpand', () => {
+    spyOn(component, 'mapUlElementToExpand');
+    component.refreshUIWithMappedElements();
+    expect(component.mapUlElementToExpand).toHaveBeenCalled();
+  });
+
+  describe('DOM UI tests', () => {
+    beforeEach(() => {
+      component.selectedNode = mockNode;
+      fixture.detectChanges();
+    });
+
+    it('should call setTreeBranchesState on expandAll button click', () => {
+      const clickMock = spyOn(component, 'setTreeBranchesState');
+      element
+        .queryAll(By.css('.btn-link'))[0]
+        .triggerEventHandler('click', null);
+      expect(clickMock).toHaveBeenCalledWith(true);
+    });
+
+    it('should call setTreeBranchesState on collapseAll button click', () => {
+      const clickMock = spyOn(component, 'setTreeBranchesState');
+      element
+        .queryAll(By.css('.btn-link'))[1]
+        .triggerEventHandler('click', null);
+      expect(clickMock).toHaveBeenCalledWith(false);
+    });
+
+    it('should render node title in the DOM', () => {
+      const rootElement: HTMLElement = element.queryAll(By.css('.node-title'))[0]
+        .nativeElement;
+      expect(rootElement.innerText).toContain(mockNode.title);
+    });
+
+    it('should display correct node children length', () => {
+      const childElement: HTMLElement = element.query(By.css('.node-title > span'))
+        .nativeElement;
+      fixture.detectChanges();
+      expect(childElement.innerText).toContain(`(${mockNode.children.length})`)
+    });
+
+    describe('testing selected node tree', () => {
+      it('should return root title', () => {
+        const domTree: HTMLElement = element.query(By.css('ul > li')).nativeElement;
+      const domElements: any = domTree.querySelectorAll('cx-generic-link');
+        expect(domElements[0].getAttribute('title')).toBe(mockNode.children[0].title); 
+      });
+      it('should return root children number 1 title', () => {
+        const domTree: HTMLElement = element.query(By.css('ul > li')).nativeElement;
+        const domElements: any = domTree.querySelectorAll('cx-generic-link');
+        expect(domElements[1].getAttribute('title')).toBe(mockNode.children[0].children[0].title);
+      });
+      it('should return sub children', () => {
+        const domTree: HTMLElement = element.query(By.css('ul > li')).nativeElement;
+        const domElements: any = domTree.querySelectorAll('cx-generic-link');
+        expect(domElements[2].getAttribute('title')).toBe(mockNode.children[0].children[0].children[0].title);
```

Comment:
This part `mockNode.children[0].children[0].children[0]` is little bit painfull ;) 
Can we declare somewhere at the top some kind of segments?

for example:
`const segment_third = mockNode.children[0].children[0].children[0];` and then just using this `segment_third` variable everywhere?

Reply:
Same for rest of segments.